scripts/cmake/vcpkg_from_sourceforge.cmake

```
@@ -0,0 +1,166 @@
+
+    if(NOT DEFINED _vdus_OUT_SOURCE_PATH)
+        message(FATAL_ERROR "OUT_SOURCE_PATH must be specified.")
+    endif()
+
+    if(NOT DEFINED _vdus_SHA512)
+        message(FATAL_ERROR "SHA512 must be specified.")
+    endif()
+
+    if(NOT DEFINED _vdus_REPO)
+        message(FATAL_ERROR "The sourceforge repository must be specified.")
+    endif()
+
+    if (_vdus_DISABLE_SSL)
+        set(URL_PROTOCOL http:)
+    else()
+        set(URL_PROTOCOL https:)
+    endif()
+    set(SOURCEFORGE_HOST ${URL_PROTOCOL}//sourceforge.net/projects)
+
+    string(FIND ${_vdus_REPO} "/" FOUND_ORG)
+    if (NOT FOUND_ORG EQUAL -1)
+        string(REGEX REPLACE ".*/" "" REPO_NAME ${_vdus_REPO})
+        string(REGEX REPLACE "/.*" "" ORG_NAME ${_vdus_REPO})
+        set(ORG_NAME ${ORG_NAME}/)
+    else()
+        set(REPO_NAME )
+        set(ORG_NAME ${_vdus_REPO}/)
+    endif()
+    
+    if (DEFINED _vdus_REF)
+        set(URL "${SOURCEFORGE_HOST}/${ORG_NAME}files/${REPO_NAME}/${_vdus_REF}/${_vdus_FILENAME}")
+    else()
+        set(URL "${SOURCEFORGE_HOST}/${ORG_NAME}${REPO_NAME}/files/${_vdus_FILENAME}")
+    endif()
+        
+    set(NO_REMOVE_ONE_LEVEL )
+    if (_vdus_NO_REMOVE_ONE_LEVEL)
+        set(NO_REMOVE_ONE_LEVEL "NO_REMOVE_ONE_LEVEL")
+    endif()
+
+    string(SUBSTRING "${_vdus_SHA512}" 0 10 SANITIZED_REF)
+
+    list(APPEND SOURCEFORGE_MIRRORS
+        cfhcable        # United States
+        pilotfiber      # New York, NY
+        gigenet         # Chicago, IL
+        versaweb        # Las Vegas, NV
+        ayera           # Modesto, CA
+        netactuate      # Durham, NC
+        phoenixnap      # Tempe, AZ
+        astuteinternet  # Vancouver, BC
+        freefr          # Paris, France
+        netcologne      # Cologne, Germany
+        deac-riga       # Latvia
+        excellmedia     # Hyderabad, India
+        iweb            # Montreal, QC
+        jaist           # Nomi, Japan
+        jztkft          # Mezotur, Hungary
+        managedway      # Detroit, MI
+        nchc            # Taipei, Taiwan
+        netix           # Bulgaria
+        ufpr            # Curitiba, Brazil
+        tenet           # Wynberg, South Africa
+    )
+    
+    foreach(SOURCEFORGE_MIRROR ${SOURCEFORGE_MIRRORS})
+        set(DOWNLOAD_URL ${URL}/download?use_mirror=${SOURCEFORGE_MIRROR})
+        message(STATUS "Trying mirror ${SOURCEFORGE_MIRROR}")
+        vcpkg_download_distfile(ARCHIVE
+            URLS "${DOWNLOAD_URL}"
+            SHA512 "${_vdus_SHA512}"
+            FILENAME "${_vdus_FILENAME}"
+            SILENT_EXIT
+        )
+        
+        if (EXISTS ${ARCHIVE})
+            set(download_success 1)
+            break()
+        endif()
+    endforeach()
+
+    if (NOT download_success)
+        message(FATAL_ERROR
+                "    \n"
+                "    Couldn't download source from all mirrors, please check your network.\n"
```

```suggestion
                "    Couldn't download source from any of the mirrors, please check your network.\n"
```

Also, it would likely be better to use `[[]]` strings here:

```
        message(FATAL_ERROR [[
    Couldn't download source from any of the sourceforge mirrors, please check your network.
    If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
    variables to "http[s]://user:password@your-proxy-ip-address:port/".
    Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
]])
```